### PR TITLE
Totally ineligible notices

### PR DESCRIPTION
**Pull request by RyanEddyIC**
# PR Checklist

Please check if your PR fulfills the following requirements
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: 
[185768811](https://www.pivotaltracker.com/n/projects/2659995/stories/185768811)

# A brief description of the changes

Current behavior:
No current notice for totally ineligible applicants.

New behavior:
A new notice will be sent when an faa application has one or more totally ineligible applicants.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:
ENV['TOTALLY_INELIGIBLE_NOTICE_IS_ENABLED']

- [ ] DC
- [X] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: app/event_source/publishers/families/notices/faa_totally_ineligible_notice_requested_publisher.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Publishers
+  module Families
+    module Notices
+      # This class will register event 'fre_notice_generation.requested'
```
Comment (nisanthyaganti9) [inline app/event_source/publishers/families/notices/faa_totally_ineligible_notice_requested_publisher.rb:6]: is this comment relevant ? or this should be `faa_totally_ineligible_notice.requested`

**Review by utkarsh7989 — DISMISSED** [review]
Looks good!

File: components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/add_eligibility_determination.rb [hunk 2]
```diff
@@ -106,6 +109,19 @@ def update_applicants(elig_d, thh_entity)
               build_member_determinations(applicant, member_determinations) if member_determinations
               applicant.save
             end
+
+            notify_totally_ineligible(application) if totally_ineligible_members.include?(true) && FinancialAssistanceRegistry.feature_enabled?(:totally_ineligible_notice)
```
Comment (nisanthyaganti9) [inline components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/add_eligibility_determination.rb:113]: Do we want to notify when at-least one member is ineligible or if all the members are ineligible ?

Comment (RyanEddyIC) [reply components/financial_assistance/app/domain/financial_assistance/operations/applications/medicaid_gateway/add_eligibility_determination.rb:113]: at-least one

**Review by nisanthyaganti9 — APPROVED** [review]

**Review by utkarsh7989 — APPROVED** [review]

**Pull request merged**